### DISCUSSION TITLE
Sending variable to data table component

### DISCUSSION THREAD

**Discussion by AndMe2** [rappasoft/laravel-livewire-tables#769] [Q&A] [answered]
Hi!
I have tried to pass a variable to datable like in normal livewire component. But it does not seem to work, any hints?
The variable should then be used in the builder function like below ($this->variable->id).

I was testing Request::segment(2) just to get the ID to test and it was working, but it would be better to pass it like a variable for many reasons.

    public function builder(): Builder
    {
        return Group::query()
        ->where('organization_id', $this->variable->id);
    }

**Comment by inmanturbo** [comment, marked as answer]
In the latest version you can pass variables through mount just like you would expect. Here is an extreme example :exploding_head: 

```php
    public ?string $dbTableName;

    public function mount($dbTableName = null)
    {
        $this->dbTableName = $dbTableName;
    }

    /**
    * The base query.
    */
    public function builder() : Builder
    {
        if(!isset($this->dbTableName)) {
            return parent::builder();
        }
        
       $model = new class extends \Illuminate\Database\Eloquent\Model {
       };

       $model->setTable($this->dbTableName);

       return $model->getQuery();
    }
```
 Keep in mind that that only certain types of properties can be persisted on re-render as per [Livewire Docs](https://laravel-livewire.com/docs/2.x/properties#important-notes)

**Comment by AndMe2** [comment]
Thanks a lot! That did it.
